File: pre_post/ComputerScience/Algorithm/LeetCode/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if nums is None:
            return []
        
        nums_dict={}
        for i in range(len(nums)):
            if target-nums[i] not in nums_dict:
                nums_dict[nums[i]]=i
            else:
                return [nums_dict[target-nums[i]],i]
                
            
            

File: pre_post/ComputerScience/Algorithm/LeetCode/1.Two_Sum.cpp
class Solution {
 public:
  vector<int> twoSum(vector<int>& nums, int target) {
    vector<int> res;
    unordered_map<int, int> hm;

    for (int i = 0; i < nums.size(); i++) {
      auto t = hm.find(target - nums[i]);
      if (t != hm.end()) {
        res.push_back(i);
        res.push_back(t->second);
        break;
      }
      // 防止重复
      hm[nums[i]] = i;
    }
    return res;
  };